processing/src/main/java/org/apache/druid/segment/join/JoinConditionAnalysis.java

```diff
@@ -133,26 +142,23 @@ public String getOriginalExpression()
    */
   public boolean isAlwaysFalse()
   {
-    return nonEquiConditions.stream()
-                            .anyMatch(expr -> expr.isLiteral() && !expr.eval(ExprUtils.nilBindings()).asBoolean());
+    return anyFalseLiteralNonEquiConditions;
```

Comment:
Why not call this `isAlwaysFalse`? (It looks like it isn't used anywhere else, and it seems to me to be easier to understand the meaning of the field if it's named after what we want it to mean.)

Reply:
yeah. I wasn't sure if isAlwaysFalse could be more complex going forward. This is clearer - it took me a while to wrap my head around what isAlwaysFalse was trying to check. Added comments, and used `isAlways...`